Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -522,13 +526,29 @@ String HTMLAttachmentElement::attachmentTitleForDisplay() const
     if (indexOfLastDot == notFound)
         return title;
 
+    auto filename = StringView(title).left(indexOfLastDot);
+    auto extension = StringView(title).substring(indexOfLastDot);
+
+    if (isWideLayout() && Layout::TextUtil::directionForTextContent(filename) == TextDirection::RTL) {
+        return makeString(
+            rightToLeftMark,
+            firstStrongIsolate,
+            leftToRightIsolate,
+            filename,
+            popDirectionalIsolate,
+            zeroWidthSpace,
+            extension,
+            popDirectionalIsolate
```

Comment:
I think this deserves some comment through example(s) explaining why we need all these isolation(s), zero width space etc (I know this patch just expands on the existing LTR behavior, but that initial commit from 2018 should have added such comments as well)